Assets/Scripts/Player/Movement/PlayerMovement.cs

```diff
@@ -1,36 +1,36 @@
 using System;
-using System.Collections;
-using FrameWork.Extensions;
 using FrameWork.GridSystem;
 using UnityEngine;
+using UnityEngine.Events;
 
 namespace Player.Movement
 {
-    public class PlayerMovement : MonoBehaviour
-    { 
-        [SerializeField, Range(1,10)] private float movementSpeed;
-        
-        private int _enumLength;
-        private int _gridIndex;
+    public class PlayerMovement : Framework.Movement
+    {
+        [SerializeField] private UnityEvent startedMoving = new();
```

Comment:
```suggestion
        [SerializeField] private UnityEvent onStartedMoving = new();
        [SerializeField] private UnityEvent onStopMoving = new();        
```
Waarom allen bij het starten een event en niet bij het stoppen. Dit kan mooi het ordering van de NPC aan roepen.